DARPA-ASKEM/TERArium
# Setup eslint (with aribnb rules), prettier, and pre commit hook

- Added `eslint` for checking style rules and error and `prettier` for formatting
- Added `husky` for triggering pre commit hook
- Added `lint-staged` that runs `eslint` and `prettier` for the staged files on commit
- Fixed linting and formatting errors for all `vue`, `ts` files

**Testing manual formatting, `yarn run format`**

1. Edit any `vue`, `ts` files
2. run `yarn format` on the project root directory or on the root of the individual package
3. Check linter checks all the error and formatter formats the edited file

**Testing pre-commit hook**
1. Edit any `vue`, `ts` files
2. run `git add . ` and `git commit` (Don't push the changes to the remote)
3. Verify that linter catches the linting error and the formatter formats committed files correctly (if linter passes)
